sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -25,10 +25,12 @@ std::vector<EventImplPtr>
 Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
   std::vector<EventImplPtr> Result;
   Command *Cmd = getCommand(Event);
-  for (const DepDesc &Dep : Cmd->MDeps) {
-    if (Dep.MDepCommand)
-      Result.push_back(Dep.MDepCommand->getEvent());
-  }
+  // We can't iterate through the empty command
+  if (Cmd)
```

Comment:
I suggest using "early-exit" instead.
```c++
Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
  Command *Cmd = getCommand(Event);
  if (!Cmd)
    return {};
  std::vector<EventImplPtr> Result;
  ...
```
NOTE: `Result` is declared after we confirmed that `Cmd` is not `nullptr`.